### PR TITLE
Use CMake-generated targets to run tests on Windows

### PR DESCRIPTION
**Pull request by steven-johnson**

### CODE REVIEW

**Comment by abadams** [comment]
Looks plausible. Did you test these targets locally? LGTM if so

File: master/master.cfg [hunk 1]
```diff
@@ -570,33 +570,19 @@ def create_win_factory(os, llvm):
...
-                     workdir = 'halide-build-Release',
-                     env = target_env,
-                     timeout = 3600,
-                     command = ['cmd', '/c',
-                                'for', '%i', 'in', '(bin\\Release\\generator*.exe)', 'do', '(%i)']))
+      for testgroup in ['correctness', 'performance', 'generators', 'tutorials']:
+        factory.addStep(
+          ShellCommand(name = testgroup,
+                       description = testgroup,
+                       locks = [performance_lock.access('counting')],
```
Comment (dsharletg) [inline master/master.cfg]: This looks like it is changing the locks from 'counting' to 'exclusive' for the performance tests. Is that intended?

Comment (abadams) [reply master/master.cfg]: Oh, good catch. The performance tests need to hold the lock in exclusive mode.

Comment (steven-johnson) [reply master/master.cfg]: Good catch, I was sloppy in my generalization.

Comment (steven-johnson) [reply master/master.cfg]: PTAL

**Comment by steven-johnson** [comment]
> Did you test these targets locally

Tested the targets on my local Windows box; the changes to buildbot specifically are of course not yet tested

**Pull request merged**